Frameworks/Foundation/NSURLProtocol_WinHTTP.mm

```diff
@@ -243,20 +244,35 @@ - (instancetype)initWithRequest:(NSURLRequest*)request
     };
 }
 
+// Helper function that executes a block on the client thread - used for client callbacks
+static void __dispatchClientCallback(NSURLProtocol_WinHTTP* protocol, void (^callbackBlock)()) {
+    if (protocol->_clientThread) {
+        [protocol performSelector:@selector(_invokeBlock:) onThread:protocol->_clientThread withObject:callbackBlock waitUntilDone:NO];
```

Comment:
it's valid/specified for block objects to have an `invoke` method, so you could use `callbackBlock performSelector:@selector(invoke) ...`